sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -248,57 +244,91 @@ private[sql] trait SQLTestUtilsBase
           !spark.sessionState.catalog.functionExists(FunctionIdentifier(functionName)),
           s"Function $functionName should have been dropped. But, it still exists.")
       }
-    }
+    )
   }
 
   /**
    * Drops temporary view `viewNames` after calling `f`.
    */
   protected def withTempView(viewNames: String*)(f: => Unit): Unit = {
-    try f finally {
-      // If the test failed part way, we don't want to mask the failure by failing to remove
-      // temp views that never got created.
-      try viewNames.foreach(spark.catalog.dropTempView) catch {
-        case _: NoSuchTableException =>
-      }
-    }
+    tryWithFinallyBlock(f)(viewNames.foreach(spark.catalog.dropTempView))
```

Comment:
Is this equivalent? Previously, we swallow only `NoSuchTableException`. Could you revert the last commit?

Reply:
They are not equivalent. I thought if an unexpected exception happened before the finally block, that exception should be thrown instead. I hope the function 'tryWithFinallyBlock' does the right thing.

Understood the concern and the last commit is just reverted. I will submit another PR for the function 'tryWithFinallyBlock'. I hope it make senses.

Reply:
Yep. Thanks. Reducing the scope of the PR is always good for review.